### ISSUE TITLE
UTF-8 in files can crash sphinx-js

### ISSUE THREAD

**Issue by joshuaavalon** [erikrose/sphinx-js#67]
Python version: 3.7.0
sphinx-js: 2.6

test.js
```javascript
const words = "現";
```

If this file is in one of the source folder, even it is not documented  it will casue
> jsdoc found no JS files in the directories ['']. Make sure js_source_path is set correctly in conf.py. It is also possible (though unlikely) that jsdoc emitted invalid JSON.

By printing the cause, it is 
> 'cp950' codec can't decode byte 0xe7 in position 86388: illegal multibyte sequence

Running Jsdoc in command line does not have any problem.

Possible fix:
```python
from subprocess import Popen, PIPE

def run_jsdoc(app):
    ...
    try:
        p = Popen(jsdoc_command, stdout=PIPE) # Write to pipe instead of temp file
    except OSError as exc:
        if exc.errno == ENOENT:
            raise SphinxError(
                '%s was not found. Install it using "npm install -g jsdoc".' % jsdoc_command_name)
        else:
            raise
    try:
        doclets = load(p.stdout) # Pipe directly to json load
    except ValueError as e:
        print(e)
        raise SphinxError(
            'jsdoc found no JS files in the directories %s. Make sure js_source_path is set correctly in conf.py. It is also possible (though unlikely) that jsdoc emitted invalid JSON.' % abs_source_paths)
```

I don't why writing to temporary file will cause a encoding error though.

**Comment by erikrose** [comment]
I had no luck reproducing this with Python 3.6 and sphinx-js cb249103ed394f670cd316ce0e9e31424cb3a341 (slightly newer than 2.6). I'll try again once 3.7 finished building.

**Comment by joshuaavalon** [comment]
@erikrose It may be related to Window. I am using Window 7 and `chcp` returns "Active code page: 950" (cp950) in CMD. This may related the cause of the problem.

**Comment by erikrose** [comment]
With Python 3.7.0, sphinx-js 2.6, and Sphinx 1.7.6, I can't reproduce it. Maybe it has to do with your locale. Are you on Windows perchance?

**Comment by erikrose** [comment]
Ah, sorry; didn't get your comment before I read mine. We used to write to a pipe as you suggest, but there were limits to how much some OSes would buffer. I think we just need to work out how to keep the local locale out of it.

**Comment by erikrose** [comment]
I tried setting my LANG and LANGUAGE environment variables to various Chinese ones—zh_HK, zh_TW, and zh_CN—but still couldn't reproduce the error. Can you tell me what line the encoding error is happening on?

**Comment by erikrose** [comment]
2 ideas to try:

1. Change 'w+' to 'w+b'.
2. Change `load(temp)` to `load(getreader('utf-8')(temp))`. (You'll need to import `getreader` from `codecs`.)

**Comment by aaclayton** [comment]
Hey @erikrose - I experienced this same problem under so I was glad to find this open issue. I was getting a decode error trying to build docs in a Windows environment under python 3.6.1 and Sphinx 1.8.1. Your first suggestion (changing the temp file mode from 'w+' to 'w+b') solved my issue. o/

**Comment by erikrose** [comment]
@aaclayton Thank you! I went ahead and changed both our jsdoc and typedoc invocations to use w+b. I don't have an easy way to test on Windows myself, but we'll ship this way in the next release and see what happens!

**Comment by erikrose** [comment]
Just released 2.7.1, which includes this fix.
